packages/EasyErrorHandler/src/Bridge/Symfony/Providers/ErrorCodesByEnumProvider.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyErrorHandler\Bridge\Symfony\Providers;
+
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesEnumInterface;
+use EonX\EasyErrorHandler\Interfaces\ErrorCodesProviderInterface;
+use ReflectionClass;
+use Symfony\Component\Finder\Finder;
+
+final class ErrorCodesByEnumProvider implements ErrorCodesProviderInterface
+{
+    private const ERROR_CODE_NAME_PREFIX = 'Error';
+
+    public function __construct(private readonly string $projectDir)
+    {
+    }
+
+    public function provide(): array
+    {
+        $enums = $this->locateErrorCodesEnums();
+
+        if (\count($enums) === 0) {
+            return [];
+        }
+
+        $errorCodes = [];
+        foreach ($enums as $enum) {
+            $cases = $enum::cases();
+            foreach ($cases as $case) {
+                if (\str_starts_with($case->name, self::ERROR_CODE_NAME_PREFIX) === true) {
+                    $errorCodes[$case->name] = $case->value;
+                }
+            }
+        }
+
+        return $errorCodes;
+    }
+
+    /**
+     * @return array<mixed>
+     */
+    private function locateErrorCodesEnums(): array
```

Comment:
What about rewrite with `nikic/php-parser`.

Chat GPT suggest like this:
```php
<?php

use PhpParser\Error;
use PhpParser\ParserFactory;

$interfaceName = 'MyInterface';

$parser = (new ParserFactory)->create(ParserFactory::PREFER_PHP7);

$directory = new RecursiveDirectoryIterator(__DIR__);
$iterator = new RecursiveIteratorIterator($directory);
$regex = new RegexIterator($iterator, '/\.php$/');

foreach ($regex as $file) {
    try {
        $code = file_get_contents($file->getPathname());
        $stmts = $parser->parse($code);

        foreach ($stmts as $stmt) {
            if ($stmt instanceof \PhpParser\Node\Stmt\Class_) {
                $implements = $stmt->implements;
                foreach ($implements as $impl) {
                    if ($impl->toString() === $interfaceName) {
                        echo "Class " . $stmt->name . " implements " . $impl->toString() . " in file " . $file->getPathname() . "\n";
                    }
                }
            }
        }
    } catch (Error $e) {
        // handle parse error
    }
}
```

Reply:
I really need to install phpParser just because chatGPT suggest it?

Reply:
Anyway done. Please check.

Reply:
We decided to continue with attributes.